assets/js/universal-dapp.js

```diff
@@ -388,7 +413,7 @@ UniversalDApp.prototype.runTx = function( data, args, cb) {
     var isConstructor = args.bytecode !== undefined;
 
     var gas = self.options.getGas ? self.options.getGas : 1000000;
-    var value = self.options.getValue ? self.options.getValue : 0;
+    var value = (args.value || (self.options.getValue ? self.options.getValue : 0)) - 0;
```

Comment:
With regards to the branch I mentioned, The idea was to have a single tx input area (same place as origin) and using the `options.getValue` handler to get the value desired by the user for the current tx.